database/spanner/spanner.go

```diff
@@ -142,9 +143,12 @@ func (s *Spanner) Run(migration io.Reader) error {
 	}
 
 	// run migration
-	stmts := migrationStatements(migr)
-	ctx := context.Background()
+	stmts, err := migrationStatements(migr)
```

Comment:
Make statement parsing optional and disabled by default.
`migrate` should not be [parsing migrations](https://github.com/golang-migrate/migrate/blob/master/MIGRATIONS.md#migration-content-format) since we treat the migration contents as an opaque blob

Not sure why this isn't already behind an optional. Let's name it something like `x-clean-statements` and put it in `Config.CleanStatements`

Reply:
Fixed and pushed. Felt that `x-strip-comments` is a little more representative of the feature. What do you think?

Reply:
I believe [ParseDDL()](https://godoc.org/cloud.google.com/go/spanner/spansql#ParseDDL) also supports multiple statements, so `x-strip-comments` isn't an accurate description of the behavior.
`CleanStatements` seems like be best name so far since it re-formats the statements. e.g. does more than just parsing and validating structure.

I'm ok with making this a backwards incompatible change since it's limited to the spanner db driver and the trade-off of cleaner code is worth it. So anyone using this updated driver will need to specify `x-clean-statements` to continue multi-statement support. We'll document this in the release notes as a breaking change.

Reply:
Seems like it didn't end up getting listed as breaking, FWIW.

Running into some issues with this after updating, mentioned in #426

Reply:
Thanks @wyardley! I've updated the [v4.12.0](https://github.com/golang-migrate/migrate/releases/tag/v4.12.0) release notes to mention this breaking change.